everestjs/everestjs.cpp

```diff
@@ -381,93 +380,99 @@ static Napi::Value boot_module(const Napi::CallbackInfo& info) {
 
         // uses[_optional] property
         auto uses_vars_prop = Napi::Object::New(env);
-        auto uses_opt_vars_prop = Napi::Object::New(env);
+        auto uses_list_vars_prop = Napi::Object::New(env);
         auto uses_cmds_prop = Napi::Object::New(env);
-        auto uses_opt_cmds_prop = Napi::Object::New(env);
+        auto uses_list_cmds_prop = Napi::Object::New(env);
         for (auto& requirement : module_manifest["requires"].items()) {
             auto const& requirement_id = requirement.key();
-            json req_route = config->resolve_requirement(module_id, requirement_id);
-            if (req_route.is_null()) {
-                continue;
+            json req_route_list = config->resolve_requirement(module_id, requirement_id);
+            // if this was a requirement with min_connections == 1 and max_connections == 1,
+            // this will be simply a single connection, but an array of connections otherwise
+            // (this array can have only one entry, if only one connection was provided, though)
+            const bool is_list = req_route_list.is_array();
+            if (!is_list) {
+                req_route_list = json::array({req_route_list});
             }
+            auto req_mod_vars_array = Napi::Array::New(env);
+            auto req_mod_cmds_array = Napi::Array::New(env);
+            for (uint64_t i = 0; i < req_route_list.size(); i++) {
```

Comment:
I changed this to uint64_t because we use this type for the Requirement index

Reply:
No, no concious decisision, uint64_t is fine